### PR TITLE
feat: add obs e2e test

### PR DESCRIPTION
**Pull request by deer-hang**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

```
Ran 2 of 2 Specs in 40.240 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
```

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: test/e2e/obs_csi_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	"github.com/huaweicloud/huaweicloud-csi-driver/test/e2e/helper"
+)
+
+var _ = ginkgo.Describe("OBS CSI STANDARD testing", func() {
+	var sc *storageV1.StorageClass
+	var pvc *corev1.PersistentVolumeClaim
+
+	ginkgo.BeforeEach(func() {
+		sc = createSCOBS()
+		pvc = createPvcOBS(sc)
```
Comment (chengxiangdong) [inline test/e2e/obs_csi_test.go:22]:
```suggestion
		pvc = createObsPvc(sc)
```

**Comment by chengxiangdong** [comment]
pls use the following command to add execute permission to the e2e script file
```
git update-index --chmod=+x .\hack\post-run-obs-e2e.sh
git update-index --chmod=+x .\hack\pre-run-obs-e2e.sh
git update-index --chmod=+x .\hack\run-obs-e2e-test.sh
```

File: hack/run-obs-e2e-test.sh [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
```
Comment (chengxiangdong) [inline hack/run-obs-e2e-test.sh:1]: this file is unnecessary, please remove it

**Comment by chengxiangdong** [comment]
E2E test failed, please check the code.

File: test/e2e/obs_csi_test.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	"github.com/huaweicloud/huaweicloud-csi-driver/test/e2e/framework"
+	"github.com/huaweicloud/huaweicloud-csi-driver/test/e2e/helper"
+)
+
+var _ = ginkgo.Describe("OBS CSI STANDARD testing", func() {
+	var sc *storageV1.StorageClass
+	var pvc *corev1.PersistentVolumeClaim
+
+	ginkgo.BeforeEach(func() {
+		sc = createSCOBS()
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply test/e2e/obs_csi_test.go:21]: done

File: test/e2e/obs_csi_test.go (hunk 1 above)
Comment (deer-hang) [reply test/e2e/obs_csi_test.go:22]: done

File: test/e2e/obs_csi_test.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+	storageV1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/huaweicloud/huaweicloud-csi-driver/pkg/common"
+	"github.com/huaweicloud/huaweicloud-csi-driver/test/e2e/framework"
+	"github.com/huaweicloud/huaweicloud-csi-driver/test/e2e/helper"
+)
+
+var _ = ginkgo.Describe("OBS CSI STANDARD testing", func() {
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply test/e2e/obs_csi_test.go:16]: done

File: test/e2e/obs_csi_test.go [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+		pvc = createPvcOBS(sc)
+	})
+
+	ginkgo.AfterEach(func() {
+		framework.RemovePVC(kubeClient, pvc.Namespace, pvc.Name)
+		framework.RemoveStorageClass(kubeClient, sc.Name)
+	})
+
+	ginkgo.It("Mount OBS to a Pod testing", func() {
+		pod := newPodWithPvcOBS(pvc)
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply test/e2e/obs_csi_test.go:31]: done

File: test/e2e/obs_csi_test.go [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+		framework.CreatePod(kubeClient, pod)
+		framework.WaitPodPresentOnClusterFitWith(kubeClient, pod.Namespace, pod.Name, func(pod *corev1.Pod) bool {
+			return pod.Status.Phase == corev1.PodRunning
+		})
+		framework.RemovePod(kubeClient, pod.Namespace, pod.Name)
+		framework.WaitPodDisappearOnCluster(kubeClient, pod.Namespace, pod.Name)
+	})
+
+	ginkgo.It("Expand OBS testing", func() {
+		pod := newPodWithPvcOBS(pvc)
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply test/e2e/obs_csi_test.go:41]: done

File: test/e2e/obs_csi_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (deer-hang) [reply test/e2e/obs_csi_test.go:31]: Duplicate name in sfsturbo-csi-test

File: test/e2e/obs_csi_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (deer-hang) [reply test/e2e/obs_csi_test.go:41]: Duplicate name in sfsturbo-csi-test

File: test/e2e/obs_csi_test.go [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+		newStorage := int64(6 * common.GbByteSize)
+		pvc.Spec.Resources.Requests = corev1.ResourceList{
+			"storage": *resource.NewQuantity(newStorage, resource.BinarySI),
+		}
+
+		framework.UpdatePVC(kubeClient, pvc)
+
+		framework.WaitPVCPresentOnClusterFitWith(kubeClient, pvc.Namespace, pvc.Name,
+			func(pvc *corev1.PersistentVolumeClaim) bool {
+				fmt.Printf("pvc.Status.Capacity.Storage(): %d\n", pvc.Status.Capacity.Storage().Value())
```
[the remark this answers is not in the document]
Comment (deer-hang) [reply test/e2e/obs_csi_test.go:57]: done

File: hack/run-obs-e2e-test.sh (hunk 2 above)
Comment (deer-hang) [reply hack/run-obs-e2e-test.sh:1]: this file will be deleted after e2e-obs shell checked.

**Pull request merged**